src/malli/generator.cljc

```diff
@@ -163,10 +164,21 @@
     (gen-one-of gs)
     (-never-gen options)))
 
+(defn- -merge-keyword-dispatch-map-into-entries [schema]
+  (let [dispatch (-> schema m/properties :dispatch)]
+    (cond-> schema
+      (keyword? dispatch)
+      (mu/transform-entries
+       #(map (fn [[k :as e]]
+               (cond-> e
+                 (not= ::m/default k)
+                 (update 2 mu/merge [:map [dispatch [:= k]]]))) %)
```

Comment:
Can `k` be nil? I think that needs to become `[:= nil nil]`.

Reply:
Ditto if `k` is a map.

Reply:
true, good catch!